lib/linkerd2-proxy-core/src/recover.rs

```diff
@@ -0,0 +1,18 @@
+use super::Error;
+use futures::Stream;
+
+/// An error recovery strategy.
+pub trait Recover<E: Into<Error> = Error> {
```

Comment:
This is perhaps overly bikesheddy, but: why `Recover` rather than `Retry`? IMO, the name `Recover` implies a broader range of error recovery behavior, like fallbacks, as well as the backoff-and-retry behavior this trait implements.

I wonder if there's a way to come up with a more generic error-recovery trait that can encompass both this and the fallback layer? Not a blocker, just a thought.

Reply:
in our context, retry is an operation that occurs on an individual request. This _is_ a broader error recovery strategy.